src/migrate.c

```diff
@@ -0,0 +1,384 @@
+        return REDISMODULE_OK;
+    }
+
+    RaftReq *req = RaftReqInit(ctx, RR_IMPORT_KEYS);
+    req->r.import_keys.term = (raft_term_t) term;
+    req->r.import_keys.migration_session_key = migration_session_key;
+
+    int num_keys = (argc -3) / 2;
+    req->r.import_keys.num_keys = num_keys;
+    req->r.import_keys.key_names = RedisModule_Calloc(num_keys, sizeof(RedisModuleString *));
+    req->r.import_keys.key_serialized = RedisModule_Calloc(num_keys, sizeof(RedisModuleString *));
+
+    for (int i = 0; i < num_keys; i++) {
+        req->r.import_keys.key_names[i] = RedisModule_HoldString(rr->ctx, argv[3 + (i*2)]);
+        req->r.import_keys.key_serialized[i] = RedisModule_HoldString(rr->ctx, argv[3 + (i*2) + 1]);
+    }
+
+    raft_entry_t *entry = RaftRedisSerializeImport(&req->r.import_keys);
+    entry->id = rand();
+    entry->type = RAFT_LOGTYPE_IMPORT_KEYS;
+    entryAttachRaftReq(rr, entry, req);
+
+    int e = raft_recv_entry(rr->raft, entry, NULL);
+    if (e != 0) {
+        replyRaftError(req->ctx, e);
+        entryDetachRaftReq(rr, entry);
+        raft_entry_release(entry);
+        goto fail;
+    }
+
+    raft_entry_release(entry);
+
+    return REDISMODULE_OK;
+
+fail:
+    RaftReqFree(req);
+    return REDISMODULE_OK;
+}
+
+static void raftAppendRaftUnlockDeleteEntry(RedisRaftCtx *rr, RaftReq *req)
+{
+    raft_entry_t *entry = RaftRedisLockKeysSerialize(req->r.migrate_keys.keys, req->r.migrate_keys.num_keys);
+    entry->id = rand();
+    entry->type = RAFT_LOGTYPE_DELETE_UNLOCK_KEYS;
+    entryAttachRaftReq(rr, entry, req);
+
+    int e = raft_recv_entry(rr->raft, entry, NULL);
+    if (e != 0) {
+        RedisModule_ReplyWithError(req->ctx, "ERR Unable to unlock/delete migrated keys, try again");
+        replyRaftError(req->ctx, e);
+        entryDetachRaftReq(rr, entry);
+        raft_entry_release(entry);
+        goto error;
+    }
+
+    raft_entry_release(entry);
+
+    /* Unless applied by raft_apply_all() (and freed by it), the request
+     * is pending so we don't free it or unblock the client.
+     */
+    return;
+
+error:
+    RaftReqFree(req);
+}
+
+static void transferKeysResponse(redisAsyncContext *c, void *r, void *privdata)
+{
+    Connection *conn = privdata;
+    JoinLinkState *state = ConnGetPrivateData(conn);
+    RedisRaftCtx *rr = ConnGetRedisRaftCtx(conn);
+    RaftReq *req = state->req;
+
+    redisReply *reply = r;
+
+    if (!reply) {
+        ConnMarkDisconnected(conn);
+        RedisModule_ReplyWithError(req->ctx, "ERR: connection dropped importing keys into remote cluster, try again");
+        RaftReqFree(req);
+    } else if (reply->type == REDIS_REPLY_ERROR) {
+        ConnAsyncTerminate(conn);
+        replyWithFormatErrorString(req->ctx, "RAFT.IMPORT failed: %.*s", (int) reply->len, reply->str);
+        RaftReqFree(req);
+    } else if (reply->type != REDIS_REPLY_STATUS || reply->len != 2 || strncmp(reply->str, "OK", 2)) {
```

Comment:
```suggestion
    } else if (reply->type != REDIS_REPLY_STATUS || reply->len != 2 || strncmp(reply->str, "OK", 2) != 0) {
```
Just to silence linter.